### DISCUSSION TITLE
How to set lsp for rust

### DISCUSSION THREAD

**Discussion by v4nderstruck** [syl20bnr/spacemacs#16030] [Q&A]
I've been trying to setup spacemacs for rust with rust-analyzer.

This is the config I added (rest is pretty much default spacemacs config)

added to the existing place in the dotspacemacs/layers function 
```lisp
   dotspacemacs-configuration-layers
   '(
     (lsp :variables
        lsp-rust-server 'rust-analyzer)
     rust

) 
```

However, after reloading Emacs still chooses to load RLS language server ....

<img width="1122" alt="image" src="https://github.com/syl20bnr/spacemacs/assets/99101223/c84c0678-4302-488a-b3d0-c2ca62a9f2c4">

Can someone help ?

**Comment by waridh** [comment]
Hi, it's been a year. I was wondering if you've managed to resolve this issue?
